File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.py
# 문제가 연속정답이면 score += 1
# 입력받은 원소들 안에서 각각 쪼개서 연산하자

import sys
input = sys.stdin.readline
N = int(input())

for i in range(N):
    OX = input()
    score = 0
    sum = 0
    for j in OX:
        if j == 'O':
            score += 1
        else:
            score = 0
        sum += score
    
    print(sum)

File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.java
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.BufferedReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            String OX = br.readLine();
            int score = 0;
            int sum = 0;

            for (char ch : OX.toCharArray()) {
                if (ch == 'O') {
                    score += 1;
                }
                else {
                    score = 0;
                }
                sum += score;
            }
            
            System.out.println(sum);
            
        }
        
        
    }
}